src/faultlog/util.cpp

```diff
@@ -182,4 +184,58 @@ json parseCallout(const std::string callout)
     return sectionJson;
 }
 
+bool isECOModeEnabled(struct pdbg_target* coreTgt)
+{
+    ATTR_ECO_MODE_Type ecoMode;
+    if (DT_GET_PROP(ATTR_ECO_MODE, coreTgt, ecoMode))
+    {
+        return false;
+    }
+
+    if (ecoMode == ENUM_ATTR_ECO_MODE_ENABLED)
+    {
+        return true;
+    }
+    return false;
+}
+
+bool isECOcore(struct pdbg_target* target)
+{
+    const char* tgtClass = pdbg_target_class_name(target);
+    if (!tgtClass)
+    {
+        return false;
+    }
+    std::string strTarget(tgtClass);
+    if (strTarget != "core" && strTarget != "fc")
+    {
+        return false;
+    }
+    if (strTarget == "core")
+    {
+        return isECOModeEnabled(target);
+    }
+    struct pdbg_target* coreTgt;
+    pdbg_for_each_target("core", target, coreTgt)
+    {
+        if (isECOModeEnabled(coreTgt))
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+std::string pdbgTargetName(struct pdbg_target* target)
+{
+    if (isECOcore(target))
+    {
+        return "Cache-Only Core";
+    }
+    if (pdbg_target_name(target) == nullptr)
+    {
+        return "";
+    }
+    return pdbg_target_name(target);
```

Comment:
`auto trgtName = pdbg_target_name(target);`
`return (trgtName ? trgtName : "");`
Above 5 lines of code can be optimized into 2 lines if we wish so :-)